pkg/operator/cluster/ceph/osd/pod.go

```diff
@@ -40,70 +44,152 @@ const (
 	locationEnvVarName        = "ROOK_LOCATION"
 )
 
-func (c *Cluster) makeDaemonSet(selection rookalpha.Selection, config rookalpha.Config) *extensions.DaemonSet {
```

Comment:
have we thought about the migration path from 0.7 to these changes? We are maintaining a manual upgrade guide from release to release that we would need to update. For example, if someone was consuming all nodes and all devices with the daemonset, how can they migrate? It would be great if we just instructed them to delete the daemonset, then our new deployments would start up with the correct osds when we notice that the devices are already in use.

This could also be the same approach for the operator to repair osds that somehow have their deployments deleted. The operator could ensure the deployment is created for previously provisioned osds.

Reply:
I believe manual migration is the best path. I can make a script for that in the next PR.

Reply:
documentation is fine for now, we don't need to automate it yet as long as it is possible to upgrade manually.